### PR TITLE
FOGL-155_vaibhav Generate HTML from docstrings for all code under src/python

### PR DESCRIPTION
**Pull request by vaibhav-ScaleDB**
**Changes:**

1. API docs are created on the fly now.
2. Added module files for current code.
3. Added support to run api docs from makefile. 
4. Added support to see changes in docstrings through livedoc (autodoc)

### CODE REVIEW

**Comment by vaibhav-ScaleDB** [comment]
Running doc-test over this PR will fail, this is not a regression and not caused by this PR. 
There were always docstring issues in our code but since this PR now adds source code docstrings in .rst files and these docstrings are not as per PEP8 guidelines, doc tests fails. In order to pass them, we need to fix the docstrings in our source code, issue https://scaledb.atlassian.net/browse/FOGL-260

**Comment by Terris-Dianomic** [comment]
This is awesome!!

**Comment by Terris-Dianomic** [comment]
@praveen Is this ready to go?

File: docs/index.rst [hunk 1]
```diff
@@ -10,6 +10,8 @@ Welcome to FogLAMP's documentation!
 
    build-tools
 
+   api/foglamp
```
Comment (praveen-garg) [inline docs/index.rst]: @Terris-Dianomic suggest the correct name `api/foglamp`

**Comment by Terris-Dianomic** [comment]
It's too bad github won't let me reply above. "FogLAMP API" sounds good to me.. Will that work?

File: src/python/requirements-dev.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
 -r requirements.txt
 -r requirements-test.txt
+-r requirements-doc.txt
```
Comment (vaibhav-ScaleDB) [inline src/python/requirements-dev.txt]: Actually Yes, for documenting the package, we need to parse all modules (.py files) and some modules are importing foglamp

File: docs/index.rst (hunk 1 above)
Comment (Terris-Dianomic) [inline docs/index.rst]: FogLAMP API?

File: src/python/requirements-dev.txt (hunk 2 above)
Comment (Terris-Dianomic) [inline src/python/requirements-dev.txt]: I mean requirements-dev.txt includes everything, even doc, which is fine

**Pull request merged**